testsuite/classlibrary/TestServerOptions.sc

```diff
@@ -0,0 +1,30 @@
+TestServerOptions : UnitTest {
+
+
+	test_numAudioChannels {
+		var options = ServerOptions.new;
+
+		var testAudio = {
+			this.assert(
+				options.numAudioBusChannels == (
+					options.numPrivateAudioBusChannels + options.numInputBusChannels + options.numOutputBusChannels
+				),
+				"sections of audio channels should add up to the total number of audio channels\nnumPrivateAudioBusChannels: %, numInputBusChannels: %, numOutputBusChannels: %\n"
+				.format(options.numPrivateAudioBusChannels, options.numInputBusChannels, options.numOutputBusChannels)
+			);
+
+		};
+
+		testAudio.value;
+		options.numAudioBusChannels = rrand(16, 1000);
+		testAudio.value;
+		options.numPrivateAudioBusChannels = rrand(16, 1000);
+		testAudio.value;
+		options.numInputBusChannels = rrand(4, 8);
+		testAudio.value;
+		options.numOutputBusChannels = rrand(4, 8);
```

Comment:
These tests would be more useful if they tested ranges of values (or maxima and minima) rather than picking randomly. Are these the full allowed ranges for these values? (Is that enforced?)